### ISSUE TITLE
tui: restore cursor at exit

### ISSUE THREAD

**Issue by martin21** [neovim/neovim#7393]
- `nvim --version`: 0.2.0
- Vim (version: ) behaves differently? no
- Operating system/version: Debian GNU/Linux Sid
- Terminal name/version: Konsole
- `$TERM`: xterm-256color

### Steps to reproduce using `nvim -u NORC`
1. Have Konsole running with a cursor that doesn´t blink, for example solid block (could be default)
1. Run `nvim -u NORC`
2. Quit the editor

### Actual behaviour
Blinking cursor in Konsole session.

### Expected behaviour
Non-blinking cursor as it was before starting nvim.

### Reasoning

I think it is good practice to return the terminal to the original state when quitting an application that changes the terminal settings. I think this is what the typical user will expect. For example when I run `dpkg-reconfigure locales` I expect it to set the background color back to my old one, black, instead of leaving it blue. Which it does.

I am aware of: neovim + konsole = blinking cursor #7138. Yet in there you fixed the wrong configuration behavior for the cursor, but not that it does not restore the blinking state of the cursor to what it was before. So i opened this new, but related issue.

**Comment by justinmk** [comment]
see https://github.com/neovim/neovim/wiki/FAQ#cursor-style-isnt-restored-after-exiting-nvim

related https://github.com/neovim/neovim/pull/5556

**Comment by egmontkob** [comment]
See also #4396 which is this + another cursor related issue already reported.

**Comment by justinmk** [comment]
duplicate of https://github.com/neovim/neovim/issues/4396
